src/libraries/System.Private.CoreLib/src/System/IO/UnmanagedMemoryAccessor.cs

```diff
@@ -491,11 +491,11 @@ public void Write(long position, decimal value)
 
             unsafe
             {
-                int* valuePtr = (int*)(&value);
-                int flags = *valuePtr;
-                int hi = *(valuePtr + 1);
-                int lo = *(valuePtr + 2);
-                int mid = *(valuePtr + 3);
+                int[] bits = decimal.GetBits(value);
```

Comment:
This is allocating an array on every access.  If we want to do that for the big endian code path for now, ok, but we can't add such allocation to the little endian code path.

Reply:
OK, changed to keep the logic and just handle endianness directly.